modo/helpers.py

```diff
@@ -226,62 +226,61 @@ def parse_region(region: str) -> tuple[str, Optional[int], Optional[int]]:
             f"Invalid region format: {region}. Expected 'chr:start-end' (start/end optional)"
         )
 
-    chrom, _, start, end = matches.groups()
+    reference_name, _, start, end = matches.groups()
     if start:
         start = int(start)
     if end:
         end = int(end.replace("-", ""))
 
-    return (chrom, start, end)
+    return (reference_name, start, end)
 
 
 def get_fileformat(path: str) -> Optional[str]:
     """Return the file format"""
     pattern = re.compile(r"\S+\.vcf(\.\w+)?")
     if path.endswith(("fasta", "fa")):
         file_format = "FASTA"
-    elif path.endswith("fastq", "fq"):
+    elif path.endswith(("fastq", "fq")):
         file_format = "FASTQ"
     elif path.endswith(".cram"):
         file_format = "CRAM"
+    elif path.endswith(".bam"):
+        file_format = "BAM"
+    elif path.endswith(".sam"):
+        file_format = "SAM"
     elif pattern.match(path):  # .vcf/.vcf.gz (or other compression)
         file_format = "VCF"
     elif path.endswith(".bcf"):
-        file_format = "BAM"
+        file_format = "BCF"
     else:
         file_format = None
     return file_format
 
 
-def bytesio_to_alignment_segments(
-    bytesio_buffer, reference_filename: str
-) -> Iterator[AlignedSegment]:
-    """Takes a BytesIO buffer and returns a pysam iterator"""
-    """Takes a BytesIO buffer and returns a pysam iterator"""
-    # Create a temporary file to write the bytesio data
-    with tempfile.NamedTemporaryFile() as temp_file:
-        # Write the contents of the BytesIO buffer to the temporary file
-        temp_file.write(bytesio_buffer.getvalue())
-
-        # Seek to the beginning of the temporary file
-        temp_file.seek(0)
-
-        # Open the temporary file as a pysam.AlignmentFile object
-        with AlignmentFile(
-            temp_file.name, "rc", reference_filename=reference_filename
-        ) as alignment_file:
-            # Iterate over the alignments in the file
-            for alignment in alignment_file:
-                yield alignment
+def make_pysam_inFile(
+    path: str, fileformat: str, reference_filename: Optional[str] = None
+):
+    """Create a pysam AlignmentFile of VariantFile"""
+    if fileformat == "CRAM":
+        pysamFile = AlignmentFile(
```

Comment:
__nitpick__: Same as above - can we stick to snake case `pysamFile` --> `pysam_file`?